pkg/tsdb/azuremonitor/types.go

```diff
@@ -90,3 +90,30 @@ type column struct {
 	Name string `json:"name"`
 	Type string `json:"type"`
 }
+
+// azureMonitorJSONQuery is the model for the JSON query sent for azure monitor requests
+// from the frontend.
+type azureMonitorJSONQuery struct {
+	AzureMonitor struct {
+		Aggregation         string  `json:"aggregation"`
+		Alias               string  `json:"alias"`
+		AllowedTimeGrainsMs []int64 `json:"allowedTimeGrainsMs"`
+		Dimension           string  `json:"dimension"`
+		DimensionFilter     string  `json:"dimensionFilter"`
+		Format              string  `json:"format"`
+		MetricDefinition    string  `json:"metricDefinition"`
+		MetricName          string  `json:"metricName"`
+		MetricNamespace     string  `json:"metricNamespace"`
+		ResourceGroup       string  `json:"resourceGroup"`
+		ResourceName        string  `json:"resourceName"`
+		TimeGrain           string  `json:"timeGrain"`
+		Top                 string  `json:"top"`
+	} `json:"azureMonitor"`
+	DatasourceID int64  `json:"datasourceId"`
```

Comment:
Need to check with properties are not use here, and remove them to avoid confusion between the mix of properties that are part of the common model, and parts that are not.